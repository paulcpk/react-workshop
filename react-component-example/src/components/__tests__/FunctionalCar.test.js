import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import React from 'react'
import Car from '../FunctionalCar'

const CAR_COLOR = 'red'

it('renders without crashing', () => {
  render(<Car />)
})

test('renders car with correct text', () => {
  render(<Car color={CAR_COLOR} />)
  const linkElement = screen.getByText(`Hi, I am a ${CAR_COLOR} Car! ðŸš™`)
  expect(linkElement).toBeInTheDocument()
})

test('renders car correctly with different text', () => {
  const color = 'yellow'
  render(<Car color={color} />)
  const linkElement = screen.getByText(`Hi, I am a ${color} Car! ðŸš™`)
  expect(linkElement).toBeInTheDocument()
})

it('renders correctly', () => {
  const tree = renderer.create(<Car color={CAR_COLOR} />).toJSON()
  expect(tree).toMatchSnapshot()
})
