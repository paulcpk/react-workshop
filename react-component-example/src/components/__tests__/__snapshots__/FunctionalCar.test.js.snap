// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "1rem",
    }
  }
>
  Hi, I am a 
  red
   Car! ðŸš™
</div>
`;
